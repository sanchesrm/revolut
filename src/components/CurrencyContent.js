import React, { Component } from "react";
import { Menu, Segment, Input, Form } from "semantic-ui-react";
import PropTypes from "prop-types";

const style = {
  borderNone: {
    border: "none"
  },
  currencyHeader: {
    fontSize: "35px",
    display: "block",
    lineHeight: "27px"
  },
  currencySubheader: { fontSize: "18px", marginTop: "20px" },
  segmentGroup: {
    backgroundColor: "rgba(255, 255, 255, 0)",
    border: "none",
    boxShadow: "none",
    color: "white"
  },
  divStyle: {
    height: "250px",
    marginTop: "25px",
    borderRadius: "25px",
    padding: "15px",
    boxShadow: "5px 10px"
  }
};

const currenciesAvailable = ["GBP", "EUR", "USD"];

class App extends Component {
  renderInput = () => {
    return (
      <input type="text" style={{ textAlign: "start", direction: "rtl" }} />
    );
  };
  render() {
    return (
      <div
        style={{
          ...style.divStyle,
          backgroundImage: `linear-gradient(to right, ${
            this.props.primaryColor
          }, ${this.props.secondaryColor})`
        }}
      >
        <Menu
          pointing
          secondary
          fluid
          widths={3}
          borderless
          inverted
          another={"first"}
          style={{ borderWidth: "0px", fontSize: "18px" }}
        >
          {currenciesAvailable.map(currency => {
            return (
              <Menu.Item
                name={currency}
                key={currency}
                active={currency === this.props.choosenCurrency}
                onClick={(undefined, { name }) =>
                  this.props.handleCurrencyChoose(name, this.props.tabIndex)
                }
              />
            );
          })}
        </Menu>
        <Segment.Group horizontal style={style.segmentGroup}>
          <Segment textAlign="left" padded="very" style={style.borderNone}>
            <div style={style.currencyHeader}>Left</div>
            <div style={style.currencySubheader}>You have: $13.00</div>
          </Segment>

          <Segment textAlign="right" padded="very" style={style.borderNone}>
            <Input
              style={{ display: "block", textAlign: "start", direction: "rtl" }}
              size="massive"
              autoFocus={this.props.autoFocus}
              transparent
              inverted
              labelPosition={"right"}
              tabIndex={this.props.tabIndex}
              input={<input style={{ textAlign: "start", direction: "rtl" }} />}
            />
            <div style={style.currencySubheader}>U$ 1.00 = L 1.00</div>
          </Segment>
        </Segment.Group>
      </div>
    );
  }
}

App.propTypes = {
  primaryColor: PropTypes.string,
  secondaryColor: PropTypes.string,
  choosenCurrency: PropTypes.string,
  autoFocus: PropTypes.bool,
  tabIndex: PropTypes.number,
  handleCurrencyChoose: PropTypes.func
};

export default App;
