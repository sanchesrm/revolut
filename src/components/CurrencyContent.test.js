import Adapter from "enzyme-adapter-react-16";
import React from "react";
import toJson from "enzyme-to-json";
import { configure, shallow, mount } from "enzyme";
import { CurrencyContent } from "./CurrencyContent";

describe("Login Component", () => {
  let props;
  beforeEach(() => {
    configure({ adapter: new Adapter() });
    props = {
      choosenCurrency: ["GBP", "USD"],
      position: 0,
      handleCurrencyChoose: jest.fn(),
      changeHandler: jest.fn(),
      inputValue: 1234,
      amount: 0.0,
      RatesReducer: { exchangesRates: { 0: 1.0, 1: 1.0 } }
    };
  });

  test("render correctly", () => {
    expect.assertions(1);

    const wrapper = shallow(<CurrencyContent {...props} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test("render correctly", () => {
    expect.assertions(1);

    props = { ...props, position: 1, RatesReducer: {} };
    const wrapper = shallow(<CurrencyContent {...props} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test("handleCurrencyChoose to have been called", () => {
    expect.assertions(1);

    const wrapper = mount(<CurrencyContent {...props} />);

    const handleCurrencyChooseMethod = jest.spyOn(
      props,
      "handleCurrencyChoose"
    );

    const currencyToggle = wrapper.find("div.three.item.menu a").get(1);
    currencyToggle.props.onClick();

    expect(handleCurrencyChooseMethod).toHaveBeenCalled();
  });

  test("handleCurrencyChoose to have been called", () => {
    expect.assertions(1);

    const wrapper = mount(<CurrencyContent {...props} />);

    const changeHandlerMethod = jest.spyOn(props, "changeHandler");

    const currencyChangeInput = wrapper
      .find("div.ui.massive.inverted.transparent.input input")
      .get(0);
    currencyChangeInput.props.onChange({
      preventDefault: () => {},
      target: {
        value: "1234"
      },
      persist: () => {}
    });

    expect(changeHandlerMethod).toHaveBeenCalled();
  });
});
