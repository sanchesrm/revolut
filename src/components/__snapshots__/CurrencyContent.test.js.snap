// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Component render correctly 1`] = `
<div
  style={
    Object {
      "backgroundImage": "linear-gradient(to right, #1c5aff, #265ed2)",
      "borderRadius": "25px",
      "boxShadow": "5px 10px",
      "height": "225px",
      "marginTop": "25px",
    }
  }
>
  <Menu
    another="first"
    borderless={true}
    fluid={true}
    inverted={true}
    pointing={true}
    secondary={true}
    style={
      Object {
        "borderWidth": "0px",
        "fontSize": "18px",
      }
    }
    widths={3}
  >
    <MenuItem
      active={true}
      key="GBP"
      name="GBP"
      onClick={[Function]}
    />
    <MenuItem
      active={false}
      key="EUR"
      name="EUR"
      onClick={[Function]}
    />
    <MenuItem
      active={false}
      key="USD"
      name="USD"
      onClick={[Function]}
    />
  </Menu>
  <SegmentGroup
    horizontal={true}
    style={
      Object {
        "backgroundColor": "rgba(255, 255, 255, 0)",
        "border": "none",
        "boxShadow": "none",
        "color": "white",
      }
    }
  >
    <Segment
      padded="very"
      style={
        Object {
          "border": "none",
        }
      }
      textAlign="left"
    >
      <div
        style={
          Object {
            "display": "block",
            "fontSize": "35px",
            "lineHeight": "27px",
          }
        }
      >
        GBP
      </div>
      <div
        style={
          Object {
            "fontSize": "18px",
            "marginTop": "20px",
          }
        }
      >
        You have:
        <span
          style={
            Object {
              "fontWeight": 600,
            }
          }
        >
           £ 0
        </span>
      </div>
    </Segment>
    <Segment
      padded="very"
      style={
        Object {
          "border": "none",
        }
      }
      textAlign="right"
    >
      <Input
        input={
          <CurrencyInput
            allowNegative={true}
            autoFocus={false}
            decimalSeparator="."
            inputType="text"
            onChange={[Function]}
            onChangeEvent={[Function]}
            precision={2}
            prefix=""
            selectAllOnFocus={false}
            suffix=""
            thousandSeparator=""
            value={1234}
          />
        }
        inverted={true}
        size="massive"
        style={
          Object {
            "width": "110px",
          }
        }
        tabIndex={1}
        transparent={true}
        type="text"
      />
      <div
        style={
          Object {
            "fontSize": "18px",
            "marginTop": "20px",
          }
        }
      >
        £ 1.00 - $ 1.00
      </div>
    </Segment>
  </SegmentGroup>
</div>
`;

exports[`Login Component render correctly 2`] = `
<div
  style={
    Object {
      "backgroundImage": "linear-gradient(to right, #265ed2, #1c5aff)",
      "borderRadius": "25px",
      "boxShadow": "5px 10px",
      "height": "225px",
      "marginTop": "25px",
    }
  }
>
  <Menu
    another="first"
    borderless={true}
    fluid={true}
    inverted={true}
    pointing={true}
    secondary={true}
    style={
      Object {
        "borderWidth": "0px",
        "fontSize": "18px",
      }
    }
    widths={3}
  >
    <MenuItem
      active={false}
      key="GBP"
      name="GBP"
      onClick={[Function]}
    />
    <MenuItem
      active={false}
      key="EUR"
      name="EUR"
      onClick={[Function]}
    />
    <MenuItem
      active={true}
      key="USD"
      name="USD"
      onClick={[Function]}
    />
  </Menu>
  <SegmentGroup
    horizontal={true}
    style={
      Object {
        "backgroundColor": "rgba(255, 255, 255, 0)",
        "border": "none",
        "boxShadow": "none",
        "color": "white",
      }
    }
  >
    <Segment
      padded="very"
      style={
        Object {
          "border": "none",
        }
      }
      textAlign="left"
    >
      <div
        style={
          Object {
            "display": "block",
            "fontSize": "35px",
            "lineHeight": "27px",
          }
        }
      >
        USD
      </div>
      <div
        style={
          Object {
            "fontSize": "18px",
            "marginTop": "20px",
          }
        }
      >
        You have:
        <span
          style={
            Object {
              "fontWeight": 600,
            }
          }
        >
           $ 0
        </span>
      </div>
    </Segment>
    <Segment
      padded="very"
      style={
        Object {
          "border": "none",
        }
      }
      textAlign="right"
    >
      <Input
        input={
          <CurrencyInput
            allowNegative={true}
            autoFocus={false}
            decimalSeparator="."
            inputType="text"
            onChange={[Function]}
            onChangeEvent={[Function]}
            precision={2}
            prefix=""
            selectAllOnFocus={false}
            suffix=""
            thousandSeparator=""
            value={1234}
          />
        }
        inverted={true}
        size="massive"
        style={
          Object {
            "width": "110px",
          }
        }
        tabIndex={2}
        transparent={true}
        type="text"
      />
      <div
        style={
          Object {
            "fontSize": "18px",
            "marginTop": "20px",
          }
        }
      >
        $ 1.00 - £ 0
      </div>
    </Segment>
  </SegmentGroup>
</div>
`;
